system/Debug/Toolbar/Collectors/History.php

```diff
@@ -81,14 +83,21 @@ public function setFiles(int $current, int $limit = 20)
 
             $contents = @json_decode($contents);
             if (json_last_error() === JSON_ERROR_NONE) {
-                preg_match_all('/\d+/', $filename, $time);
-                $time = (int) end($time[0]);
+                preg_match_all('/debugbar_(.*)\.json/s', $filename, $time, PREG_SET_ORDER);
+                $time = number_format((float) end($time[0]), 4, '.', '');
```

Comment:
This can be simplified to use `preg_match`

Reply:
Done